src/persist-client/src/lib.rs

```diff
@@ -383,14 +368,17 @@ impl PersistClient {
                 Arc::clone(&self.metrics),
             )
         });
-        let shard_upper = machine.fetch_upper().await;
+        let writer_id = WriterId::new();
+        let (shard_upper, _) = machine.register_writer(&writer_id, system_time()).await;
```

Comment:
oh, do we need to mess with persist's clock from tests outside persist? it's easy to plumb in down via PersistConfig if we do

Reply:
If you think the NowFn stuff doesn't fit here then ignore this. I had a light suspicion that it would be nice for the codebase to have exactly one place where we call `SystemTime::now()`, but if it doesn't fit for your needs, whatever.

Reply:
this PR is a blocker for another PR that's a blocker for the cloud billing work that jeannette is about to start, so I have a light preference to just get this merged however paul can most easily get it passing CI. it really should be straightforward to plumb via PersistConfig though, so I'm happy to do that as a fast follow myself